packages/docs/docs/animation-utils/use-interpolate-styles.md

```diff
@@ -0,0 +1,63 @@
+---
+image: /generated/articles-docs-animation-utils-use-interpolate-styles.png
+title: useInterpolateStyles()
+id: use-interpolate-styles
+crumb: "@remotion/animation-utils"
+---
+
+_Part of the [`@remotion/animation-utils`](/docs/animation-utils) package._
+
+This hook provides a convenient way to interpolate styles based on a specified range of frames or seconds, allowing for smooth animations between different styles.
+
+## API
+
+### `useInterpolateStyles(options: InterpolateStylesParams): Style`
+
+#### Parameters
+
+- **inputRangeInFrames**: _Optional_. An array of frame steps for interpolation.
+- **inputRangeInSeconds**: _Optional_. An array of duration steps in seconds for interpolation.
+- **outputStyles**: _Required_. An array of styles to be interpolated.
+
+### Return value
+
+- A style object representing the interpolated styles based on the current frame.
+
+### Example
+
+```tsx twoslash
+// ---cut---
+import {
+  useInterpolateStyles,
+  makeTransform,
+  translateY,
+} from "@remotion/animation-utils";
+
+const MyComponent: React.FC = () => {
+  const animatedStyles = useInterpolateStyles({
```

Comment:
It's a nice idea! But this makes it less flexible unfortunately: You have to obey by the rules of hooks (no conditional rendering, cannot .map() over them), and there is no control over the timing, everything seems to be linear timing. You also cannot use it outside a component or with a spring animation. The API should work more like interpolate:

```tsx
const styles = interpolateStyles(frame, [0, 30, 60], [{opacity: 0}, {opacity: 1}, {opacity: 0}])
```

So it doesn't need to be a hook.

Reply:
It also means the function cannot be used outside of Remotion, which could be useful. I'd like if we scrap this function entirely and just ship a "pure" function